:PROPERTIES:
:ID:       b084016b-83cb-4d36-b6ca-afd90d8bc900
:END:
#+title: Error Handling
#+filetags: :Javascript:

- tags :: [[id:98730b92-6677-4ef0-bf88-3c8cf7a33504][Javascript]]

* throw, try, catch

if there is error in the try block, code in catch block will be evaluated.

#+begin_src js
// these two expressions have the same functionality.
const negNum = new Error("factorial does not support negative numbers");
const negNum = Error("factorial does not support negative numbers");

function factorial(n) {
  if (n < 0) {
    throw negNum;
  }
  let res = 1;
  while (n > 0) {
    res = res * n;
    n--;
  }
  return res;
}

let n = -10;
try {
  console.log(factorial(n));
} catch {
  console.log(factorial(-n));
}
#+end_src
